sklearn/multioutput.py

```diff
@@ -0,0 +1,218 @@
+from .utils import check_random_state
+from .utils.estimator_checks import multioutput_estimator_convert_y_2d
+from .utils.validation import _num_samples
+from .utils.validation import check_consistent_length
+from .utils.validation import check_is_fitted
+from .externals.joblib import Parallel
+from .externals.joblib import delayed
+from .externals import six
+
+from sklearn.multiclass import OneVsRestClassifier
+
+
+def _check_estimator(estimator):
+    """Make sure that an estimator implements the necessary methods."""
+    if (not hasattr(estimator, "score") and
+            not hasattr(estimator, "predict_proba")):
+        raise ValueError("The base estimator should implement "
+                         "score or predict_proba!")
+
+
+class MultiOutputClassifier(ClassifierMixin):
+    """
+    Converts a single target classifer estimator into
+    a multi-target classifier estimator.
+
+    This class fits and predicts a series of one-versus-all models
+    to response matrix Y, which has n_samples and p_target variables,
+    on the predictor Matrix X with n_samples and m_feature variables.
+    This allows for multiple target variable classifications. For each
+    target variable (column in y), a separate estimator is fit.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        An estimator object implementing `fit` and `predict_proba`.
+
+    n_jobs : int, optional, default: 1
+        The number of jobs to use for the computation. If -1 all CPUs are used.
+        If 1 is given, no parallel computing code is used at all, which is
+        useful for debugging. For n_jobs below -1, (n_cpus + 1 + n_jobs) are
+        used. Thus for n_jobs = -2, all CPUs but one are used.
+        The number of jobs to use for the computation.
+
+        Note that parallel processing only occurs if there are multiple classes
+        within each target variable.
+        It does each target variable in y in series.
+
+    Attributes
+    __________
+    estimator: Sklearn estimator:
+        The base estimator used to construct the model.
+
+    """
+
+    def __init__(self, estimator=None, n_jobs=1):
+
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """ Fit the model to data.
+        Creates a seperate model for each Response column.
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        y : (sparse) array-like, shape = [n_samples, n_outputs]
+            Multi-output targets. An indicator matrix turns on multilabel
+            classification.
+
+        Returns
+        -------
+        self
+        """
+
+        _check_estimator(self.estimator)
+        # check to see that the X and y have the same number of rows.
+        check_consistent_length(X, y)
+
+        # Calculate the number of classifiers
+        self._n_outputs = y.shape[1]
+
+        self.estimators_ = {}
```

Comment:
Sorry but this was the initial idea and I hadn't changed it since then. I suppose it was to maintain the order. Would it be fine to change it to list ?